Developer.md

```diff
@@ -0,0 +1,28 @@
+# Building and Testing the Java Client Core Library
+
+## Build/Dev Environment Prerequisites 
+ - [Java version 8+ e.g Temurin distribution](https://projects.eclipse.org/projects/adoptium.temurin/downloads)
```

Comment:
Temurin is not a requirement for development. It's what's recommended for us to run the integration tests on GitHub. We prefer Temurin over OpenJDK because it has better support (like backporting security patches to older JDKs).

The developer can use whatever JDK they like.